sdk/servicebus/azure-servicebus/azure/servicebus/aio/management/_management_client_async.py

```diff
@@ -0,0 +1,853 @@
+        """List the subscriptions of a ServiceBus Topic Runtime Information.
+
+        :param Union[str, TopicDescription] topic: The topic that owns the subscription.
+        :rtype: ItemPaged[~azure.servicebus.management.SubscriptionRuntimeInfo]
+        """
+        try:
+            topic_name = topic.name  # type: ignore
+        except AttributeError:
+            topic_name = topic
+
+        def entry_to_subscription(entry):
+            subscription = SubscriptionRuntimeInfo._from_internal_entity(entry.content.subscription_description)
+            subscription.name = entry.title
+            return subscription
+
+        extract_data = functools.partial(
+            extract_data_template, SubscriptionDescriptionFeed, entry_to_subscription
+        )
+        get_next = functools.partial(
+            get_next_template, functools.partial(self._impl.list_subscriptions, topic_name), **kwargs
+        )
+        return AsyncItemPaged(
+            get_next, extract_data)
+
+    async def get_rule(
+            self, topic: Union[str, TopicDescription], subscription: Union[str, SubscriptionDescription],
+            rule_name: str, **kwargs) -> RuleDescription:
+        """Get a topic subscription rule
+
+        :param Union[str, TopicDescription] topic: The topic that owns the subscription.
+        :param Union[str, SubscriptionDescription] subscription: The subscription that owns the rule.
+        :param str rule_name: Name of the rule.
+        :rtype: ~azure.servicebus.management.RuleDescription
+        """
+        try:
+            topic_name = topic.name  # type: ignore
+        except AttributeError:
+            topic_name = topic
+        try:
+            subscription_name = subscription.name  # type: ignore
+        except AttributeError:
+            subscription_name = subscription
+        entry_ele = await self._get_rule_element(topic_name, subscription_name, rule_name, **kwargs)
+        entry = RuleDescriptionEntry.deserialize(entry_ele)
+        if not entry.content:
+            raise ResourceNotFoundError(
+                "Rule('Topic: {}, Subscription: {}, Rule {}') does not exist".format(
+                    subscription_name, topic_name, rule_name))
+        rule_description = RuleDescription._from_internal_entity(entry.content.rule_description)
+        return rule_description
+
+    async def create_rule(
+            self, topic: Union[str, TopicDescription], subscription: Union[str, SubscriptionDescription],
+            rule: Union[str, RuleDescription], **kwargs) -> RuleDescription:
+        """Create a subscription of a topic.
+
+        :param Union[str, TopicDescription] topic: The topic that will own the to-be-created subscription rule.
+        :param Union[str, SubscriptionDescription] subscription: The subscription that will own the to-be-created rule.
+        :param Union[str, ~azure.servicebus.management.RuleDescription] rule: The rule name or a
```

Comment:
Shouldn't all the objects be fully qualified?

Reply:
what is a rule just with a rule_name?
TBH I would expect create_rule to only accept RuleDescription

Reply:
changed to use fully qualified names for this place and all other places.

@yunhaoling removed the str. It should be a RuleDescription